### PR TITLE
fix/fees bittrex

### LINKED ISSUE #2084: [BUG] Filled order fees are not recorded correctly
**Describe the bug**

Fees for orders that are filled are not recorded correctly for many connectors, instead of taking fees reported from exchange's API, the connectors call `c_get_fee` (which estimates fee). This results in incorrect fees stored on trades table (sqlite) and incorrect performance reported on `history` command.

This mistake was partly due to Binance connector had this bug which other connectors copied the code from. The bug was fixed on 29 December 2020 on Binance connector so connectors created after the date should not get affected.

Connectors which broadcast fee incorrectly on `OrderFilledEvent`:

- [ ] ~Kucoin~
- [x] Binance Perpetual Derivative
- [x] Bifinex
- [x] Coinbase Pro
- [x] Bittrex
- [ ] ~OkEx~
- [x] Huobi
- [ ] ~Beaxy~
- [x] FTX
- [ ] ~Blocktane~
- [ ] ~Coinzoom~
- [ ] ~DigiFinex~
- [x] Gate.io
- [ ] Liquid

┆Issue is synchronized with this [Clickup task](https://app.clickup.com/t/1pd1e7e) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by aarmoa**
**Before submitting this PR, please make sure**:

- [x] Your code builds clean without any errors or warnings
- [x] You are using approved title ("feat/", "fix/", "docs/", "refactor/")

**A description of the changes proposed in the pull request**:
The PR include the required changes to consider the fees informed by the exchange when a trade takes place, instead of estimating them.
Partially closes #2084 

**Tests performed by the developer**:
Added new unit tests for the order fill process and the fees registration

**Tips for QA testing**:
Please check that partial and complete fills are working correctly in Bittrex connector. Also verify if the fees are now being registered correctly.

### CODE REVIEW

**Comment by nikspz** [comment]
Can't `connect bittrex` on that branch
Bot is showing error `No module named 'hummingbot.loggerß'`

[logs_hummingbot.log](https://github.com/CoinAlpha/hummingbot/files/7653314/logs_hummingbot.log)
image
image
image

**Comment by aarmoa** [comment]
@nikspz sorry for the inconvenience. There was a typo in one import statement. I have fixed it.

**Comment by rxlxrxsx** [comment]
- Filled orders are not being tracked on history, bottom nav and data files (csv/sqlite)

Screen Shot 2021-12-05 at 11 31 21 AM

Logs and config: [bittrex.zip](https://github.com/CoinAlpha/hummingbot/files/7655401/bittrex.zip)

**Comment by RC-13** [comment]
Re-tested the PR and we are still seeing the issue with recording the filled trades
bittrex-first-fill-not-recorded1

[Bittrex-fills-and-logs.zip](https://github.com/CoinAlpha/hummingbot/files/7665837/Bittrex-fills-and-logs.zip)

File: hummingbot/connector/exchange/bittrex/bittrex_in_flight_order.pyx [hunk 1]
```diff
@@ -26,6 +26,9 @@ cdef class BittrexInFlightOrder(InFlightOrderBase):
             initial_state
         )
 
+        self.trade_id_set = set()
+        self.fee_asset = self.quote_asset
```
Comment (CrimsonJacket) [inline hummingbot/connector/exchange/bittrex/bittrex_in_flight_order.pyx:30]: I don't think it is right to assume that the fee asset is the quote asset. We should deduce this from the order side.

**Review by CrimsonJacket — CHANGES_REQUESTED** [review]
One comment on the default 'fee_asset' in BittrexInFlightOrder. Other than that LGTM

File: hummingbot/connector/exchange/bittrex/bittrex_in_flight_order.pyx (hunk 1 above)
Comment (aarmoa) [reply hummingbot/connector/exchange/bittrex/bittrex_in_flight_order.pyx:30]: The current estimated fee is always calculated as an amount in quote asset. I could not find any place in the documentation from Bittrex that suggest the fee is charged in the base asset. If we intend to change that we should have some validation from Bittrex, either in the form of documentation or by their confirmation.

**Comment by aarmoa** [comment]
@RC-13 and @rxlxrxsx I have added some changes to allow the connector to wait for fill events before processing the complete events (in order to not lose the fill events). If you have a moment please retest the PR with the new logic.

**Review by PtrckM — APPROVED** [review]
Reported Issues are now fixed. LGTM, approving in behalf of QA

**Comment by nikspz** [comment]
Is that okay that Bittrex shows `Fail to submit order according to Websocket API` error when it's `insufficient balance to place` an order?
image

**Comment by aarmoa** [comment]
Hello @nikspz. That happens when the condition is detected by the exchange. It could mean that the connector is not checking the balance in the account before submitting the order. We should open an issue if we want to investigate that.

**Review by CrimsonJacket — APPROVED** [review]
Changes LGTM.

**Pull request merged**